ts: limit the recording of child metrics to a small set of metrics

jasonlmfong opened the pull request:
This change adds an explicit list of metrics which we perform child metrics collection for.
This helps bound the performance impact even when the cluster setting `timeseries.child_metrics.enabled` is turned on.

Epic: CRDB-55079
Release: The list of metrics which we record child metrics in TSDB has been reduced from all changefeed metrics to a defined list.

dhartunian reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158986)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @alyshanjahani-crl and @jasonlmfong)

___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/158986#-OfybzTs5BDmmR1Ev2Sv:-OfybzTs5BDmmR1Ev2Sw:b-uubxxh):*
nit: note which cluster setting.
___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/158986#-OfybvzXBhcyGUjOv2pp:-OfybvzXBhcyGUjOv2pq:b-9cvr0v):*
nit: `Release note: None`


<!-- Sent from Reviewable.io -->

jasonlmfong reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158986#-:-OfyeMJA8-4HU444jxUE:b-814g8k)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @alyshanjahani-crl)



<!-- Sent from Reviewable.io -->

jasonlmfong committed:
ts: limit the recording of child metrics to a small set of metrics

This change adds an explicit list of metrics which we perform child metrics collection for.
This helps bound the performance impact even when the cluster setting `timeseries.child_metrics.enabled` is turned on.

Epic: CRDB-55079
Release: The list of metrics which we record child metrics in TSDB has been reduced from all changefeed metrics to a defined list.
